cluster/src/main/java/org/apache/iotdb/cluster/utils/RaftUtils.java

```diff
@@ -302,4 +317,106 @@ public static Status createErrorStatus(String errorMsg){
     status.setCode(-1);
     return status;
   }
+
+  public static ConcurrentHashMap<String, PeerId> getGroupLeaderCache() {
+    return groupLeaderCache;
+  }
+
+  public static Map<Integer, String> getPhysicalRing() {
+    SortedMap<Integer, PhysicalNode> hashNodeMap = router.getPhysicalRing();
+    Map<Integer, String> res = new LinkedHashMap<>();
+    hashNodeMap.entrySet().forEach(entry -> res.put(entry.getKey(), entry.getValue().getIp()));
+    return res;
+  }
+
+  public static Map<Integer, String> getVirtualRing() {
+    SortedMap<Integer, VirtualNode> hashNodeMap = router.getVirtualRing();
+    Map<Integer, String> res = new LinkedHashMap<>();
+    hashNodeMap.entrySet().forEach(entry -> res.put(entry.getKey(), entry.getValue().getPhysicalNode().getIp()));
```

Comment:
It can be more concise for all your entry `foreach`, for example:
```suggestion
    hashNodeMap.forEach((key, value) -> res.put(key, value.getPhysicalNode().getIp()));
```

Reply:
cool, fixed